spec/function-builder.spec.ts

```diff
@@ -132,7 +147,39 @@ describe('FunctionBuilder', () => {
       functions
         .region('asia-northeast1')
         .runWith({ timeoutSeconds: 600, memory: '256MB' });
-    }).to.throw(Error, 'TimeoutSeconds');
+    }).to.throw(Error, 'RuntimeOptions.timeoutSeconds');
+  });
+
+  it('should throw an error if user chooses a failurePolicy of type number', () => {
+    expect(() =>
+      functions.runWith({
+        failurePolicy: (1234 as unknown) as boolean,
+      })
+    ).to.throw(Error, 'RuntimeOptions.failurePolicy');
+  });
+
+  it('should throw an error if user chooses a failurePolicy of type string', () => {
+    expect(() =>
+      functions.runWith({
+        failurePolicy: ('string-value' as unknown) as boolean,
+      })
+    ).to.throw(Error, 'RuntimeOptions.failurePolicy');
+  });
+
+  it('should throw an error if user chooses a failurePolicy.retry of type number', () => {
```

Comment:
Same as above - we can just have one test here that tests error being thrown when `retry` is not an object.

Reply:
4059b00a73e8136c7dfbb51e234b0495d784c31d